var/spack/repos/builtin/packages/py-horton/package.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import platform
+
+
+class PyHorton(PythonPackage):
+    """HORTON: Helpful Open-source Research TOol for N-fermion systems. Copyright (C) 2011-2016 The HORTON Development Team
+       For more information, visit HORTON's website: http://theochem.github.com/horton/latest
+    """
+
+    # FIXME: Add a proper url for your package's homepage here.
+    homepage = "https://theochem.github.io/horton"
+    url      = "https://github.com/theochem/horton/releases/download/2.1.1/horton-2.1.1.tar.gz"
+
+    maintainers = ['frobnitzem']
+
+    version('2.1.1', sha256='4b3f87920d881030ba80f097326a744de2cfee5316aa4499cc9a6501f64b5060')
+
+    depends_on('python@2.7:2.99', type=('build', 'run'))
+    # note: The py-setuptools dep should be type='build', but build_systems/python.py
+    # adds "args += ['--single-version-externally-managed', '--root=/']"
+    # This conflicts with the present setuptools version (required by python2).
+    # So, it's hacked to run-dep here.
+    depends_on('py-setuptools@:44.1.0',   type='run')
```

Comment:
I don't see anywhere where setuptools is needed at build-time or run-time. I think you can remove this dependency.

Reply:
If I remove this line, I get the error
```
$ spack spec py-chem-horton

==> Error: An unsatisfiable version constraint has been detected for spec:
    python@2.7.17%gcc@8.4.0+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib arch=linux-ubuntu18.04-skylake

while trying to concretize the partial spec:

    py-setuptools@50.3.2%gcc@8.4.0 arch=linux-ubuntu18.04-skylake

py-setuptools requires python version 3.5:, but spec asked for 2.7.17
```
So the setuptools dependency is being added somehow by spack.  I removed the line, but the package no longer builds.

Reply:
Oh, so it's because it's a transitive dependency. The new concretizer fixes that bug. Honestly it's easier to just not use Python 2.

Reply:
This package only exists in python2.